### PR TITLE
Parser/publio

### PR DESCRIPTION
**Pull request by sledzgrzegorz**
This PR contains parser for book on promotion under url: 
http://www.publio.pl/szukaj,promocja.html?sections=EBOOK

### CODE REVIEW

File: parser/src/main/java/parser/UrlParser.java [hunk 1]
```diff
@@ -12,7 +14,12 @@
  */
 public abstract class UrlParser {
 
+    protected static final Logger logger = LogManager.getRootLogger();
```
Comment (caroo7) [inline parser/src/main/java/parser/UrlParser.java:17]: why static!!!

File: parser/src/main/java/parser/publio/NextPageUrlConsumer.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * This thread is needed for consuming url to next pages to produce urls to books details
+ */
+class NextPageUrlConsumer implements Runnable {
+
+    private static final Logger logger = LogManager.getRootLogger();
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/NextPageUrlConsumer.java:18]: why static!!!

File: parser/src/main/java/parser/publio/PublioUrlCrawler.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        Thread nextPageUrlProducerThread = new Thread(nextPageUrlProducer);
+        Thread nextPageUrlConsumerThread = new Thread(nextPageUrlConsumer);
+        nextPageUrlProducerThread.start();
+        nextPageUrlConsumerThread.start();
+
+        try {
+            nextPageUrlConsumerThread.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/PublioUrlCrawler.java:50]: maybe use log.error

File: parser/src/main/java/parser/UrlParser.java (hunk 1 above)
Comment (LazyScoundrel) [reply parser/src/main/java/parser/UrlParser.java:17]: That is why we have Lombok! ;)

File: parser/src/main/java/parser/publio/PublioUrlCrawler.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    private final PageLoader pageLoader;
+    private final PublioUrlParser publioUrlParser;
+
+    public PublioUrlCrawler(PageLoader pageLoader) {
+        this.pageLoader = pageLoader;
+        this.publioUrlParser = new PublioUrlParser();
+    }
+
+    /**
+     * This method uses NextPageUrlProducer and NextPageUrlConsumer because page publio.pl is slow then producer-consumer
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/PublioUrlCrawler.java:31]: "slower" not "slow"

File: parser/src/main/java/parser/publio/PublioUrlParser.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Created by Grzesiek on 2016-08-27.
+ */
+public class PublioUrlParser extends UrlParser {
+
+    String getLinkToNextPage(Optional<Document> document) {
+        logger.debug("document: "+document.get().baseUri());
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/PublioUrlParser.java:16]: Can be null

File: parser/src/main/java/parser/publio/PublioBookParser.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            return document.get().select("ins.product-card-price-promotion").first().text();
+        }
+        return null;
+
+    }
+
+    @Override
+    public String getPercentageDiscount(Optional<Document> document) {
+        if (document.isPresent()) {
+            //25,90 zł
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/PublioBookParser.java:46]: Move format to java doc comment

File: parser/src/main/java/parser/publio/NextPageUrlProducer.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jsoup.nodes.Document;
+import parser.PageLoader;
+
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+
+/**
```
Comment (caroo7) [inline parser/src/main/java/parser/publio/NextPageUrlProducer.java:11]: :+1:

**Pull request closed without merging**